web/models/habit.ex

```diff
@@ -30,34 +29,23 @@ defmodule Habits.Habit do
   @doc """
   The current streak is the number of consecutive daily check-ins
   for a habit up until yesterday, or today if you’ve checked in today.
-
-  Hat tip: stackoverflow.com/q/22142028/
   """
   def get_current_streak(habit) do
-    yesterday_string =
-      Habits.Date.today
-      |> Habits.Date.shift_days(-1)
-      |> Date.to_iso8601
-
-    sql = "
-      SELECT COUNT(check_ins.date)
-      FROM check_ins
-      WHERE check_ins.habit_id = $1
-      AND check_ins.date > (
-        SELECT calendar.day
-        FROM generate_series('2010-01-01'::date, '#{yesterday_string}'::date, '1 day') calendar(day)
-        LEFT OUTER JOIN check_ins
-          ON check_ins.date = calendar.day
-          AND check_ins.habit_id = $1
-        WHERE check_ins.date IS NULL
-        ORDER BY calendar.day DESC
-        LIMIT 1
-      );
-    "
+    habit
+    |> assoc(:streaks)
+    |> where([s], s.end >= ^Habits.Date.yesterday)
+    |> select([s], s.length)
```

Comment:
I thought the binding syntax (used here) would be equivalent with the bindingless syntax, e.g.

```elixir
|> select([:length])
```

but they return different results. The binding syntax returns the bare value of the selected column, while the bindingless syntax returns an entire `%Streak{}` struct with only the `select`ed attribute present.